pkg/controller/controller.go

```diff
@@ -343,13 +355,27 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		// if ScaleUpThresholdPercent is our "max target" or "slack capacity"
 		// we want to add enough nodes such that the maxPercentage cluster util
 		// drops back below ScaleUpThresholdPercent
-		nodesDelta, err = calcScaleUpDelta(untaintedNodes, cpuPercent, memPercent, cpuRequest, memRequest, nodeGroup)
+		nodesDelta, err = calcScaleUpDelta(
+			untaintedNodes,
+			cpuPercent,
+			memPercent,
+			*podRequests.Total.GetCPUQuantity(),
+			*podRequests.Total.GetMemoryQuantity(),
+			nodeGroup)
 		if err != nil {
 			log.Errorf("Failed to calculate node delta: %v", err)
 			return nodesDelta, err
 		}
 	}
 
+	if nodeGroup.Opts.ScaleOnStarve &&
+		((!podRequests.LargestPendingCPU.IsEmpty() && podRequests.LargestPendingCPU.MilliCPU > nodeCapacity.LargestAvailableCPU.MilliCPU) ||
```

Comment:
What happens if a pod with a larger request than the node size is requested? Will this just cause infinite scaling? The Kube scheduler won't allow the pod to be scheduled, but not sure if Escalator will see this pending pod and just keep scaling.

Reply:
Infinite scaling is definitely a possibility - and could even occur prior to this change. We rely on the `max_nodes` configuration option and the ASG max sizing to put an upper limit on how many nodes are created. 

Generally you would want to ensure that the AWS instances sizes you are using will accommodate the largest pods that will be scheduled to prevent this.

Reply:
node size is also another of those options that isn't immediately available here, which makes this a little tricky to add in. That does remind me that we should update the docs in `docs/configuration/nodegroup.md` to point out that this option exists, and specify the risk associated with doing this.

Reply:
We could use the `Allocatable` CPU/memory, since that's definitely available (and what actually is important anyway). The logic might be a little tricky if we want to account for an edge case of something like the largest requested pod is too large, but the 2nd largest is ok and still requires a scale up.

It might also be worth calling out in the docs that using a `LimitRange` is probably a good idea to prevent this type of scenario.